# Analyse des réponses PIX

On commence par récupérer toutes les réponses depuis la base.

In [1]:
from pixutils.airtable import lookup_challenges

lookup_challenges('Einstein')

ID: rec3UBGIRfIOkqw5S
Sur la page Wikipédia française d’Albert Einstein, où peut-on consulter la liste des modifications apportées par les contributeurs ?

ID: recW9YPNhOs8U9ud1
Qui a modifié la page Wikipédia française d’Albert Einstein le 23 mars 2016 à 01h21 ?



In [2]:
from pixutils.db import get_answers

# ID de l'épreuve
challenge_id = "rectN26toxkJmt9S4" # perrières
# rectN26toxkJmt9S4 : perrières
# rec6PW03oyqtPZ9rK : pomme & arbre

get_answers(challenge_id, 'pomme-arbre.csv')

On slugifie chaque réponse et on compte les plus fréquentes.

In [3]:
from pixutils.stats import build_analytics

results = build_analytics('pomme-arbre.csv')

1956 abandons
385 réponses distinctes parmi les 6399 réponses données



réponse    ok   ko  aberrants
0                              rue-de-la-couteauderie  2956    9          0
1                                     la-couteauderie   384    0          0
2                                        couteauderie   118    0          0
3                          rue-rue-de-la-couteauderie     0  100          0
4                               rue-de-la-coutauderie    14   50          0
5                                              num1-2     0    0         60
6                               rue-de-la-couteaudrie    11   48          0
7                                 rue-de-couteauderie     0   52          0
8                                      la-coutauderie     0   22          0
9                                   rue-des-perrieres     0   21          0
10                                rue-la-couteauderie     1   15          0
11                                 de-la-couteauderie    14    0          0
12  proverbe-la-pomme-ne-tombe-jamais-loin-de-l-arbre     0    0         14
13                                                        0   12          1
14                                     la-couteaudrie     0   10          0
15                                             num1-8     0    0          9
16                                   rue-de-montresor     0    8          0
17                                          perrieres     0    7          0
18                                         nombre-136     0    0          7
19                                   rue-couteauderie     4    2          0

In [4]:
from pixutils.science import get_clusters

get_clusters(results)

# Type -1
191 réponses dont :


réponse  ok  ko  aberrants
0                                 rue-de-montresor   0   8          0
1                                       nombre-136   0   0          7
2                                       votes-5826   0   0          5
3                                 proverbe-pommier   0   0          5
4                             boulevard-de-l-ouest   0   4          0
5                                        montresor   0   4          0
6                                   rue-des-allocs   0   4          0
7   proverbe-la-pomme-ne-tombe-pas-loin-de-l-arbre   0   0          4
8                            nomfichier-flower-jpg   0   0          4
9                                   rue-de-la-paix   0   3          0
10                                    rue-du-8-mai   0   3          0
11                                 rue-de-debouche   0   3          0
12                              la-grande-metairie   0   3          0
13                         proverbe-pomme-et-arbre   0   0          3
14                            proverbe-pomme-arbre   0   0          3

# Type 0
104 réponses dont :


réponse    ok   ko  aberrants
0       rue-de-la-couteauderie  2956    9          0
1              la-couteauderie   384    0          0
2                 couteauderie   118    0          0
3   rue-rue-de-la-couteauderie     0  100          0
4        rue-de-la-coutauderie    14   50          0
5        rue-de-la-couteaudrie    11   48          0
6          rue-de-couteauderie     0   52          0
7               la-coutauderie     0   22          0
8          rue-la-couteauderie     1   15          0
9           de-la-couteauderie    14    0          0
10              la-couteaudrie     0   10          0
11            rue-couteauderie     4    2          0
12      rue-de-la-cauteauderie     0    6          0
13                 couteaudrie     0    5          0
14      rue-de-la-couteaudiere     0    5          0

# Type 1
56 réponses dont :


réponse  ok  ko  aberrants
0             0  12          1
1        38   0   3          3
2         d   0   3          0
3    mairie   0   2          0
4     dijon   0   2          0
5        37   0   2          0
6     paris   0   2          0
7       aaa   0   2          0
8    tresor   0   2          0
9         e   0   1          0
10  montcul   0   1          0
11   tamere   0   1          0
12    prout   0   1          0
13       22   0   1          0
14     truc   0   1          0

# Type 2
7 réponses dont :


réponse  ok  ko  aberrants
0  rue-des-perrieres   0  21          0
1  rue-des-peupliers   0   1          0
2     rue-de-perries   0   1          0
3   rue-des-perriers   0   1          0
4   rue-des-arrieres   0   1          0
5     rue-de-perrier   0   1          0
6   rue-de-perrieres   0   1          0

# Type 3
5 réponses dont :


réponse  ok  ko  aberrants
0  perrieres   0   7          0
1   perriere   0   1          0
2   derriere   0   1          0
3  pierriere   0   1          0
4  parrieres   0   1          0

# Type 4
5 réponses dont :


réponse  ok  ko  aberrants
0  rue-de-ta-mere   0   2          0
1  rue-de-ma-mere   0   1          0
2    rue-dela-mer   0   1          0
3  rue-de-ma-bite   0   1          0
4  rue-de-la-gare   0   1          0

# Type 5
10 réponses dont :


réponse  ok  ko  aberrants
0  num1-2   0   0         60
1  num1-8   0   0          9
2  numb-1   0   0          6
3  num1-1   0   0          5
4  num4-1   0   0          4
5  numa-3   0   0          4
6  num1-4   0   0          3
7  numa-2   0   0          2
8  numd-1   0   0          1
9  num1-3   0   0          1

# Type 6
7 réponses dont :


réponse  ok  ko  aberrants
0  proverbe-la-pomme-ne-tombe-jamais-loin-de-l-arbre   0   0         14
1  proverbe-la-pomme-ne-tombe-jamais-loin-de-son-...   0   0          6
2   proverbe-la-pomme-ne-tombe-jamais-loin-de-larbre   0   1          2
3     proverbe-la-pomme-ne-tombe-jamais-loin-d-arbre   0   0          1
4  proverbe-les-pomme-ne-tonbe-jamais-loin-de-l-a...   0   0          1
5  proverbe-une-pomme-ne-tombe-jamais-loin-de-l-a...   0   0          1
6    proverbe-la-pomme-ne-tombe-jamis-loin-de-l-arbe   0   0          1

Pour info, pour la rue des Perrières j'ai obtenu :

# Type 0 (FAUX)
56 réponses dont :
- 
- 38
- c
- laval
- a
- e
- montcul
- 633
- mon
- et
- ggg
- 1
- mont
- rue
- rue-rue

# Type 1 (~  rue de la Couteauderie, VRAI)
104 réponses dont :
- rue-de-la-courtauderiez
- la-coutaudrie
- rue-de-la-couteuderie
- rue-de-la-courtauderie
- couteaudrie
- rue-de-la-cuteauderie
- la-couteaudene
- de-la-coutauderie
- cautederie
- la-couteaudrerie
- rue-couteauderie
- rue-de-ma-couteauderie
- rue-de-couteuderie
- la-coutedrie
- la-couteaderie

# Type 2 (~ rue des Perrières, FAUX)
7 réponses dont :
- rue-des-perriers
- rue-de-perrier
- rue-de-perries
- rue-de-perrieres
- rue-des-peupliers
- rue-des-arrieres
- rue-des-perrieres

# Type 3 (FAUX)
10 réponses dont :
- numa-3
- num1-1
- num1-3
- numd-1
- numa-2
- num1-4
- num1-2
- num1-8
- num4-1
- numb-1

# Type 4 (VRAI)
7 réponses dont :
- proverbe-la-pomme-ne-tombe-jamais-loin-de-son-arbre
- proverbe-une-pomme-ne-tombe-jamais-loin-de-l-arbre
- proverbe-la-pomme-ne-tombe-jamais-loin-de-l-arbre
- proverbe-la-pomme-ne-tombe-jamais-loin-de-larbre
- proverbe-la-pomme-ne-tombe-jamis-loin-de-l-arbe
- proverbe-les-pomme-ne-tonbe-jamais-loin-de-l-arbre
- proverbe-la-pomme-ne-tombe-jamais-loin-d-arbre

# Type 5 (~ Perrières, FAUX)
5 réponses dont :
- pierriere
- perriere
- derriere
- perrieres
- parrieres

# Type 6 (FAUX)
5 réponses dont :
- rue-de-ma-bite
- rue-de-ma-mere
- rue-dela-mer
- rue-de-ta-mere
- rue-de-la-gare

# Type -1 (réponses à l'ouest)
191 réponses dont :
- boulevard-de-l-ouest
- marthyrs
- proverbe-un-pommier
- sans-deconner-comment-je-peux-deviner-sans-tricher
- rennes
- rue-des-vignerons
- chassenolles
- jupilles
- rue-de-montresor
- rue-de-l-hermitage
- rue-du-marche
- nom-spam
- 43-grande-rue
- hfdhfg
- montresor

## Hélas parmi les réponses à l'ouest il y a des réponses qu'on pourrait considérer comme bonnes :

- rue-de-la-couteauderie-37460-montresor
- sur-la-rue-couteauderie